src/main/resources/graphql/service/quiz.graphqls

```diff
@@ -116,28 +116,289 @@ type MultipleChoiceQuestion implements Question {
     """
     Optional hint for the question, can be markdown.
     """
-    hint: String
+    hint: ResourceMarkdown
 }
 
 type MultipleChoiceAnswer {
     """
     Text of the answer, can be markdown.
     """
-    text: String!
+    text: ResourceMarkdown!
     """
     Whether the answer is correct or not.
     """
     correct: Boolean!
     """
     Feedback for when the user selects this answer, can be markdown.
     """
-    feedback: String
+    feedback: ResourceMarkdown
+}
+
+type ClozeQuestion implements Question {
+    """
+    The elements of the cloze question.
+    """
+    clozeElements: [ClozeElement!]!
+    """
+    Addtional wrong answers for the blanks.
+    """
+    additionalWrongAnswers: [String!]!
+    """
+    All selectable answers for the blanks (computed). This contains the correct answers as well as wrong answers.
+    """
+    allBlanks: [String!]!
+    """
+    Whether the blanks must be answered in free text or by selecting the correct answer from a list.
+    """
+    showBlanksList: Boolean!
+
+    # inherited from Question
+    """
+    Unique identifier of the question.
+    """
+    id: UUID!
+    """
+    Number of the question, i.e., the position of the question in the list of questions.
+    Only relevant if questionPoolingMode is ORDERED.
+    """
+    number: Int!
+    """
+    Type of the question.
+    """
+    type: QuestionType!
+    """
+    Optional hint for the question, can be markdown.
+    """
+    hint: ResourceMarkdown
+}
+
+
+interface ClozeElement {
+    """
+    Type of the element.
+    """
+    type: ClozeElementType!
```

Comment:
why have a type here when its an interface anyway?

Reply:
good question. Usually it makes it a bit easier for us on the java side, because we need some field/column in the database for the type anyway. We can remove it if you want